hieradocs.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+#  hiera::key::name:
+#    datatype: String.
+#    description: |
+#      String describing the key.
+#    found_in:
+#    - someFile.yaml
+
+SAVEIFS=$IFS
+IFS=$(echo -en "\n\b")
+
+PATTERN_PARAM=\[[*]\([^*]*\)[*]\]
+PATTERN_COMMENT=\(.+\)
+
+HIERA_DIR=../hieradata
+
+for file in `git grep -l class manifests/ | sort`;
+do
+
+    CLASS=`grep class $file | awk '{print $2}'`
+    PARAM=""
+    DOCS=""
+
+    # look only at comments that begin a line.  strip off the leading # and spaces for simplification
+    for line in `cat $file | grep ^# | sed 's/^#[ ]*//g'` ;
+    do
+        if [[ $line =~ $PATTERN_PARAM ]] ;
+        then
+            if [ "$PARAM" != "" ] ;
+            then
+                # remove leading white space and double periods (anywhere)
+                DOCS=`echo $DOCS | sed 's/^[\t .]*//g' | sed 's/\.\././g'`
+                echo "${PARAM}-A#  ${PARAM}:"
+                echo "${PARAM}-B#    datatype: String."
```

Comment:
Is the intention that eventually other datatypes will be supported? If not, why is String assumed?